frontend/src/App.tsx

```diff
@@ -74,12 +44,17 @@ function App() {
       setCurrentCards(nextCards)}
   }
 
-  function handleInputClick (id, type, value) {
+  function handleInputClick (id: CardState['id'], type: keyof CardState, value: InputValue) {
     const currentCardState = currentCards.find(card => card.id === id)
-    const newCardState = new CardState(currentCardState.id, currentCardState.shape, currentCardState.color, currentCardState.fill, currentCardState.count)
-    newCardState[type] = value
-    const newCards = currentCards.map((card) => card.id === id ? 
-    newCardState : card ) 
+
+    if (currentCardState === undefined) {
+      throw new Error('Unable to find card state')
+    }
+
+    const newCardState = new CardState(currentCardState?.id, currentCardState?.shape, currentCardState?.color, currentCardState?.fill, currentCardState?.count);
+    (newCardState as MapishCardState)[type] = value
```

Comment:
I made `MapishCardState` because we're using the instance of `CardState` as a map here. Typescript gets grumpy when we use classes as maps like this - even though JavaScript is totally cool with it. This happens because of the Type system in TypeScript; it prefers the usage of `.` when dealing with classes. 

But just because TS gets grumpy doesn't mean it's _wrong_. This is a completely fine (and even nifty) use of the class here and it's something you'd come across in a real live codebase. 

Secondly, we're casting `newCardState` as a `MapishCardState` here; that's what the `as` keyword does. It's a part of TypeScript, not JavaScript. If you're unfamiliar, it basically forces TypeScript is treat the thing on the left side as the type on the right side. This is useful when you (the developer) know better than the type system. Think of it as an escape hatch for you in the type system.